third_party/txt/src/minikin/Layout.cpp

```diff
@@ -240,7 +240,7 @@ static hb_position_t harfbuzzGetGlyphHorizontalAdvance(hb_font_t* /* hbFont */,
                                                        void* /* userData */) {
   MinikinPaint* paint = reinterpret_cast<MinikinPaint*>(fontData);
   float advance = paint->font->GetHorizontalAdvance(glyph, *paint);
-  return 256 * advance + 0.5;
+  return 256 * advance;
```

Comment:
I could not figure out why there is a `+ 0.5`. I have removed it. This constant seems to have no positive effect on the layout and would break a large suite of tests for no discernible reason.

Reply:
I will be putting the 0.5 back in, as the framework tests/goldens will fully pass with this constant addition. This may be to mirror harfbuzz's implementation. In any case, with the `+ 0.5`, we no longer need to manually roll, though it will require many more small adjustments throughout `txt_unittests`

Reply:
the 0.5 was to make the rounding round both up and down instead of flooring